### PR TITLE
Added endpoint config topic

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Content based on https://help.stackrox.com/docs/configure-stackrox/configure-endpoints/

Preview at https://openshift-docs-git-endpoint-config-gnelson.vercel.app/openshift-acs/master/configuration/configure-endpoints.html

### CODE REVIEW

File: modules/configure-endpoints-existing.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//
+// * configuration/configure-endpoints.adoc
+:_module-type: PROCEDURE
+[id="configure-endpoints-existing_{context}"]
+= Configuring endpoints for an existing instance
+
+You can configure endpoints for an existing instance of {product-title}.
+
+.Procedure
+. Download the existing _ConfigMap_:
```
Comment (bergerhoffer) [inline modules/configure-endpoints-existing.adoc:11]: Should be written as either "config map" or "`Config Map` object"

File: modules/configure-endpoints-existing.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+----
+$ oc -n stackrox create cm central-endpoints --from-file=endpoints.yaml=<directory-path>/central-endpoints.yaml -o yaml --dry-run | \
+oc label -f - --local -o yaml app.kubernetes.io/name=stackrox | \
+oc apply -f -
+----
+. Restart Central.
+
+[NOTE]
+====
+If you expose any additional ports, other than the default port `8443`, you must create network policies that allow traffic on those additional ports.
```
Comment (bergerhoffer) [inline modules/configure-endpoints-existing.adoc:30]: First comma should be removed

File: modules/configure-endpoints-new-install.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// * configuration/configure-endpoints.adoc
+:_module-type: PROCEDURE
+[id="configure-endpoints-new-install_{context}"]
+= Configuring endpoints during a new installation
+
+When you install {product-title}, by using the `roxctl` CLI, it creates a folder named `central-bundle`, which contains the necessary YAML manifests and scripts to deploy Central.
```
Comment (bergerhoffer) [inline modules/configure-endpoints-new-install.adoc:8]: First comma should be removed

File: modules/configure-endpoints-new-install.adoc [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+.Procedure
+. After you generate the `central-bundle`, open the `./central-bundle/central/02-endpoints-config.yaml` file.
+. In this file, add your custom YAML configuration under the `data:` section of the key `endpoints.yaml`.
+Make sure that you maintain a 4 space indentation for the YAML configuration.
+. Continue the installation instructions as usual.
+{product-title} uses the specified configuration.
+
+[NOTE]
+====
+If you expose any additional ports, other than the default port `8443`, you must create network policies that allow traffic on those additional ports.
```
Comment (bergerhoffer) [inline modules/configure-endpoints-new-install.adoc:19]: First comma should be removed

File: modules/custom-yaml-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
+// Module included in the following assemblies:
+//
+// * configuration/configure-endpoints.adoc
+:_module-type: CONCEPT
+[id="custom-yaml-configuration_{context}"]
+= Custom YAML configuration
+
+[role="_abstract"]
+{product-title} uses the a YAML configuration as a _ConfigMap_ object, making configurations easier to change and manage.
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:9]: * "the a" typo
* `ConfigMap` should be in backticks, not italics

File: modules/custom-yaml-configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+When you use the custom YAML configuration file, you can configure the following for each endpoint:
+
+* The protocols to use, such as `HTTP`, `gRPC`, or both.
+* Enable or disable TLS.
+* Specify server certificates.
+* Client Certificate Authorities (CA) to trust for client authentication.
+* Specify if client certificate authentication (`mTLS`) is required.
+
+You can use the configuration file to specify endpoints either during the installation or on an existing instance of {product-title}.
+However, if you expose any additional ports, other than the default port `8443`, you must create network policies that allow traffic on those additional ports.
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:20]: Second comma should be removed

File: modules/custom-yaml-configuration.adoc [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Following is a sample `endpoints.yaml` configuration file for {product-title}:
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:22]: Following -> The following

File: modules/custom-yaml-configuration.adoc [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        # Enforce TLS client authentication. If unset, do not enforce, only request certificates
+        # opportunistically.
+        required: true <8>
+        # Which TLS client CAs to serve, possible values are 'service' (CA for service
+        # certificates that {product-title} generates) and 'user' (CAs for PKI auth providers). If unset or empty, assume both.
+        certAuthorities: # if not set, assume ["user", "service"] <9>
+          - service
+----
+<1> Use `true` to disable exposure on the default port number `8443`.
+The default value is `false`, changing it to `true` might break existing functionality.
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:59]: Use a semicolon instead of a comma

File: modules/custom-yaml-configuration.adoc [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+<4> Protocols to use for the specified endpoint.
+Acceptable values are `http` and `grpc`.
+If you do not specify a value, Central listens to both HTTP and gRPC traffic on the specified port.
+If you want to expose an endpoint exclusively for the RHACS portal, use `http`.
+However, you will not be able to use the endpoint for service-to-service communication or for the `roxctl` CLI, because these clients require both gRPC and HTTP.
+Red Hat recommends that you do not specify a value of this key, to enable both HTTP and gRPC protocols for the endpoint.
+If you want to restrict an endpoint to {product-title} services only, use the *clientAuth* option.
+<5> Use it to specify the TLS settings for the endpoint.
+If you do not specify a value, {product-title} enables TLS with the default settings for all the following nested keys.
+<6> Use `true` to disable TLS on the specified endpoint.
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:76]: The field this matches up to isn't a boolean field, it's a list. Is the description wrong or is the callout on the wrong line?

Edit: Looking at the original stackrox docs, I think the <6> callout is meant for the disable field on line 39.

File: modules/custom-yaml-configuration.adoc [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+The default value is `false`.
+When you set it to `true`, you cannot specify  values for `serverCerts` and `clientAuth`.
+<7> Specify a list of sources from which to configure server TLS certificates.
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:79]: Yeah this one looks like it matches up as the description for the `serverCerts` field. Not sure what the previous line was referencing. But if you can double check that each of these are matched up properly.

File: modules/custom-yaml-configuration.adoc [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+Acceptable values are:
+* `default`: use the already configured custom TLS certificate if it exists.
+* `service`: use the internal service certificate that  {product-title} generates.
++
+The `serverCerts` list is order-dependent, it means that the first item in the list determines the certificate which Central uses by default, when there is no matching SNI (Server Name Indication).
```
Comment (bergerhoffer) [inline modules/custom-yaml-configuration.adoc:84]: which -> that

File: modules/custom-yaml-configuration.adoc (hunk 9 above)
Comment (gaurav-nelson) [reply modules/custom-yaml-configuration.adoc:77]: 🙇🏼  Thank you. I've rechecked and fixed.

**Review by bergerhoffer — COMMENTED** [review]
Just a few cleanup things, then this is ready to be merged.

**Comment by gaurav-nelson** [comment]
Thanks @bergerhoffer I've fixed recent issues and rebased this one.

**Review by bergerhoffer — COMMENTED** [review]
LGTM!

**Pull request merged**